root/components/FormRowText.js

```diff
@@ -13,34 +13,66 @@ import FieldErrors from './FieldErrors';
 import FormRow from './FormRow';
 import FormLabel from './FormLabel';
 
-type Props = {
+type InputOnChange = (SyntheticEvent<HTMLInputElement>) => void;
+
+type InputProps = {
+  defaultValue?: string,
+  +id: string,
+  +name: string,
+  onChange?: InputOnChange,
+  +required: boolean,
+  +size: ?number,
+  +type: string,
+  value?: string,
+};
+
+type CommonProps = {
   +field: ReadOnlyFieldT<?string>,
   +label: string,
-  +onChange?: (SyntheticEvent<HTMLInputElement>) => void,
   +required?: boolean,
   +size?: number,
   +type?: string,
 };
 
-const FormRowText = ({
-  field,
-  label,
-  required = false,
-  size,
-  type = 'text',
-}: Props) => (
-  <FormRow>
-    <FormLabel forField={field} label={label} required={required} />
-    <input
-      defaultValue={field.value || ''}
-      id={'id-' + field.html_name}
-      name={field.html_name}
-      required={required}
-      size={size}
-      type={type}
-    />
-    <FieldErrors field={field} />
-  </FormRow>
-);
+export type Props =
+  | $ReadOnly<{
+      ...CommonProps,
+      onChange: InputOnChange,
+      uncontrolled?: false,
+    }>
+  | $ReadOnly<{
+      ...CommonProps,
+      uncontrolled: true,
+    }>;
+
+const FormRowText = (props: Props) => {
+  const field = props.field;
+  const required = props.required ?? false;
+
+  const inputProps: InputProps = {
+    id: 'id-' + field.html_name,
+    name: field.html_name,
+    required,
```

Comment:
I assume this is because it's already the same name as the variable, but wouldn't it be more clear to do required: required? (I'm probably just too OCD)

Reply:
Maybe :) I'm just a lazy typer and used to the short syntax. If it's clearer to have it consistent with the other keys I can change it